piccolo_api/crud/endpoints.py

```diff
@@ -871,7 +889,16 @@ async def detail(self, request: Request) -> Response:
         try:
             row_id = self.table._meta.primary_key.value_type(row_id)
         except ValueError:
-            return Response("The ID is invalid", status_code=400)
+            for i in self.table._meta._foreign_key_references:
+                target = i._meta.params["target_column"]
+                if target is not None:
+                    reference_target_pk = (
+                        await self.table.select(self.table._meta.primary_key)
+                        .where(target == row_id)
+                        .first()
+                        .run()
+                    )
+                    row_id = reference_target_pk[self.table._meta.primary_key]
```

Comment:
We can do this slightly easier:

```python
reference_target_pk = (
    await self.table.select(self.table._meta.primary_key)
    .where(target == row_id)
    .output(as_list=True)
    .first()
    .run()
)[0]
```

Reply:
Unfortunately this does not work if we have multiple non-primary keys per single table.